src/controllers/role.controller.js

```diff
@@ -0,0 +1,41 @@
+import responseHandlers from '../utils/responseHandlers';
+import statusCodes from '../utils/statusCodes';
+import customMessages from '../utils/customMessages';
+import roles from '../services/role.service';
+import authentication from '../services/authentication.service';
+import utils from '../utils/authentication.utils';
+
+const { decodeToken } = utils;
+const { assignRole } = roles;
+const { findUserByEmailOrUsername } = authentication;
+const { userNotAllowedForAction, notExistUser, roleAssigned, existingRole } = customMessages;
+const { unAuthorized, forbidden, ok } = statusCodes;
+
+/**
+   * @description user role controller class
+   */
+export default class UserRoleController {
+      /**
+   * @param {Request} req Node/express request
+   * @param {Response} res Node/express response
+   * @returns {Object} Custom response depending on function's inputs
+   * @description update user role
+   */
+    static assignRole = async (req, res) => {
+        const token = req.headers.authorization.split(' ')[1];
```

Comment:
Since you are taking the decoded token from `req.sessionUser` then this line is useless